### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by davea**
- Never cache `/uprn` responses
- Add support for a read-only DB replica
- Django 4 upgrade

### CODE REVIEW

**Comment by sagepe** [comment]
Just a thought, and possibly not needed for this stage, but I wondered how easy it would be to support [libpq connection string URIs](https://www.postgresql.org/docs/13/libpq-connect.html#LIBPQ-CONNSTRING), that way we could autodetect which replica was currently the writeable one without it being explicitly specified in config.

File: mapit_labour/urls.py [hunk 1]
```diff
@@ -13,20 +13,20 @@
 format_end = r"(?:\.(?P<format>html|json))?"
 
 urlpatterns = [
-    url(r"^uprn/(?P<uprn>[0-9]+)%s$" % format_end, uprn, name="mapit_labour-uprn"),
-    url(r"^addressbase$", addressbase, name="mapit_labour-addressbase"),
-    url(r"^import/csv/?$", import_csv, name="mapit_labour-import_csv"),
-    url(
+    re_path(r"^uprn/(?P<uprn>[0-9]+)%s$" % format_end, uprn, name="mapit_labour-uprn"),
+    re_path(r"^addressbase$", addressbase, name="mapit_labour-addressbase"),
```
Comment (dracos) [inline mapit_labour/urls.py]: You could also import path and make these use the simpler one, e.g. this one is `path('addressbase')`

**Review by dracos — CHANGES_REQUESTED** [review]
Think the github action might need an apt-get update first to get that running again.
Other than that, and assuming tests then okay, only one suggestion of nicer Django URL routing :)

**Review by dracos — APPROVED** [review]
⬆️

**Pull request merged**